sql/core/src/test/resources/sql-tests/inputs/postgreSQL/groupingsets.sql

```diff
@@ -0,0 +1,563 @@
+order by 2,1;
+select ten, grouping(ten) from onek
+group by cube(ten) having grouping(ten) > 0
+order by 2,1;
+-- [SPARK-29703] grouping() can only be used with GroupingSets/Cube/Rollup
+-- select ten, grouping(ten) from onek
+-- group by (ten) having grouping(ten) >= 0
+-- order by 2,1;
+
+-- FILTER queries
+-- [SPARK-27986] Support Aggregate Expressions with filter
+-- select ten, sum(distinct four) filter (where string(four) ~ '123') from onek a
+-- group by rollup(ten);
+
+-- More rescan tests
+-- [SPARK-27877] ANSI SQL: LATERAL derived table(T491)
+-- select * from (values (1),(2)) v(a) left join lateral (select v.a, four, ten, count(*) from onek group by cube(four,ten)) s on true order by v.a,four,ten;
+-- [SPARK-27878] Support ARRAY(sub-SELECT) expressions
+-- select array(select row(v.a,s1.*) from (select two,four, count(*) from onek group by cube(two,four) order by two,four) s1) from (values (1),(2)) v(a);
+
+-- [SPARK-29704] Support the combinations of grouping operations
+-- Grouping on text columns
+-- select sum(ten) from onek group by two, rollup(string(four)) order by 1;
+-- select sum(ten) from onek group by rollup(string(four)), two order by 1;
+
+-- hashing support
+
+-- Ignore a PostgreSQL-specific option
+-- set enable_hashagg = true;
+
+-- failure cases
+
+-- Since this test is implementation specific for plans, it passes in Spark
+select count(*) from gstest4 group by rollup(unhashable_col,unsortable_col);
+-- [SPARK-27878] Support ARRAY(sub-SELECT) expressions
+-- select array_agg(v order by v) from gstest4 group by grouping sets ((id,unsortable_col),(id));
+
+-- simple cases
+
+-- [SPARK-29698] Support grouping function with multiple arguments
+-- select a, b, grouping(a,b), sum(v), count(*), max(v)
+select a, b, grouping(a), grouping(b), sum(v), count(*), max(v)
+  from gstest1 group by grouping sets ((a),(b)) order by 3,4,1,2 /* 3,1,2 */;
+-- explain (costs off) select a, b, grouping(a,b), sum(v), count(*), max(v)
+--   from gstest1 group by grouping sets ((a),(b)) order by 3,1,2;
+
+-- [SPARK-29698] Support grouping function with multiple arguments
+-- select a, b, grouping(a,b), sum(v), count(*), max(v)
+select a, b, grouping(a), grouping(b), sum(v), count(*), max(v)
+  from gstest1 group by cube(a,b) order by 3,4,1,2 /* 3,1,2 */;
+-- explain (costs off) select a, b, grouping(a,b), sum(v), count(*), max(v)
+--   from gstest1 group by cube(a,b) order by 3,1,2;
+
+-- shouldn't try and hash
+-- explain (costs off)
+--   select a, b, grouping(a,b), array_agg(v order by v)
+--     from gstest1 group by cube(a,b);
+
+-- unsortable cases
+-- [SPARK-29708] Different answers in aggregates of multiple grouping sets
+select unsortable_col, count(*)
+  from gstest4 group by grouping sets ((unsortable_col),(unsortable_col))
+  order by string(unsortable_col);
+
+-- mixed hashable/sortable cases
+select unhashable_col, unsortable_col,
+       -- [SPARK-29698] Support grouping function with multiple arguments
```

Comment:
Shall we move line 395 before line 394 and remove the indentation?

Reply:
ok